components/Cards/src/MDCCollectionViewCardCell.m

```diff
@@ -0,0 +1,209 @@
+                                           options:MDFTextAccessibilityOptionsNone];
+  self.selectedImageTintColor = checkColor;
+}
+
+- (UIColor *)backgroundColor {
+  return self.cardView.backgroundColor;
+}
+
+- (void)setCornerRadius:(CGFloat)cornerRadius {
+  self.layer.cornerRadius = cornerRadius;
+  self.cardView.cornerRadius = cornerRadius;
+  [self setNeedsLayout];
+}
+
+- (CGFloat)cornerRadius {
+  return self.cardView.cornerRadius;
+}
+
+- (void)selectionState:(MDCCardCellSelectionState)state withAnimation:(BOOL)animation {
+  _selectionState = state;
+  self.selectionMode = YES;
+  switch (state) {
+    case MDCCardCellSelectionStateSelected: {
+      if (animation) {
+        _inkAnimating = YES;
+        [self.cardView.inkView startTouchBeganAnimationAtPoint:self.lastTouch completion:nil];
+      } else {
+        if (!_inkAnimating) {
+          [self.cardView.inkView cancelAllAnimationsAnimated:NO];
+          [self.cardView.inkView addInkSublayerWithoutAnimation];
+        }
+        _inkAnimating = NO;
+        self.selectedImageView.hidden = NO;
+      }
+      [(MDCShadowLayer *)self.cardView.layer setElevation:
+       [self.cardView shadowElevationForState:MDCCardViewStateNormal]];
+
+      break;
+    }
+    case MDCCardCellSelectionStateUnselected: {
+      if (animation) {
+        [self.cardView.inkView startTouchEndedAnimationAtPoint:self.lastTouch completion:nil];
+      } else {
+        [self.cardView.inkView cancelAllAnimationsAnimated:NO];
+      }
+      [(MDCShadowLayer *)self.cardView.layer setElevation:
+       [self.cardView shadowElevationForState:MDCCardViewStateNormal]];
+      self.selectedImageView.hidden = YES;
+      break;
+    }
+  }
+}
+
+- (void)setSelectedImage:(UIImage *)selectedImage {
+  [self.selectedImageView setImage:selectedImage];
+}
+
+- (UIImage *)selectedImage {
+  return self.selectedImageView.image;
+}
+
+- (void)setSelectedImageTintColor:(UIColor *)selectedImageTintColor {
+  [self.selectedImageView setTintColor:selectedImageTintColor];
+}
+
+- (UIColor *)selectedImageTintColor {
+  return self.selectedImageView.tintColor;
+}
+
+- (void)setSelected:(BOOL)selected {
+  [super setSelected:selected];
+  if (self.selectionMode) {
+    if (selected) {
+      [self selectionState:MDCCardCellSelectionStateSelected withAnimation:NO];
+    } else {
+      [self selectionState:MDCCardCellSelectionStateUnselected withAnimation:NO];
+    }
+  }
+}
+
+#pragma mark - UIResponder
+
+- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
+  [super touchesBegan:touches withEvent:event];
+
+  UITouch *touch = [touches anyObject];
+  CGPoint location = [touch locationInView:self];
+  self.lastTouch = location;
+  if (self.selectionMode) {
+    if (!self.selected) {
+      [self selectionState:MDCCardCellSelectionStateSelected withAnimation:YES];
+    }
+  } else {
+    [_cardView setStyleForState:MDCCardViewStateHighlighted withLocation:location];
```

Comment:
Can you add a few comments to explain what's goin on here? The card will change its state if the cell has `selectionMode = NO`?